File: spy.cpp
#include<iostream>
using namespace std;
int digitsum(int n)
{
	int sum = 0;
	int num = n;
	while(num > 0)
	{
		sum += num%10;
		num /= 10;
	}
	return sum;
}
int digitproduct(int n)
{
	int prod = 1;
	int num = n;
	while(num > 0)
	{
		prod *= num%10;
		num /= 10;
	}
	return prod;
}
bool isSpy(int n)
{
	if(digitsum(n) == digitproduct(n))return true;
	return false;
}
int main()
{
	int n;
	cout<<"Enter a number : ";
	cin>>n;
	if(isSpy(n))
	{
		cout<<"Is a Spy Number."<<endl;
	}
	else
	{
		cout<<"Not a Spy Number."<<endl;
	}
}

File: spy.py
def digitsum(n:int)->int:
    sum = 0
    num = n
    while(num>0):
        sum += num%10
        num //= 10
    return sum

def digitproduct(n:int)->int:
    prod = 1
    num = n
    while(num>0):
        prod *= num%10
        num //= 10
    return prod

def isSpy(n:int)->bool:
    if digitsum(n) == digitproduct(n):
        return True
    return False

if __name__ == "__main__":
    n = int(input("Enter a number: "))
    if(isSpy(n)):
        print("Is a Spy number.")
    else:
        print("Not a Spy number.")